File: 1. foundation-part/week-02/day-2/data-structures/TelephoneBook.java
import java.util.HashMap;
import java.util.Map.Entry;

public class TelephoneBook {
	public static void main(String[] args) {
		HashMap<String, String> phoneBook = new HashMap<>();
		phoneBook.put("William A. Lathan", "405-709-1865");
		phoneBook.put("John K. Miller", "402-247-8568");
		phoneBook.put("Hortensia E. Foster", "606-481-6467");
		phoneBook.put("Amanda D. Newland", "319-243-5613");
		phoneBook.put("Brooke P. Askew", "307-687-2982");

		System.out.println(phoneBook.get("John K. Miller"));
		String value = "307-687-2982";
		for (Entry<String, String> entry : phoneBook.entrySet()) {
			if (entry.getValue() == value) {
				System.out.println(entry.getKey());
				break;
			}
		}
		if (phoneBook.containsKey("Chris E. Myer")) {
			System.out.print("yes");
		} else System.out.print("no");
	}
}

File: greenfox/4. orientation-part/js-basics/telephone-book.js
const contacts = new Map([
	["William A. Lathan", "405-709-1865"],
	["John K. Miller", "402-247-8568"],
	["Hortensia E. Foster", "606-481-6467"],
	["Amanda D. Newland", ""],
	["Brooke P. Askew", "307-687-2982"],
]);

console.log("What is John K. Miller's phone number?");
console.log(contacts.get("John K. Miller"));
console.log("");

console.log("Whose phone number is 307-687-2982?");
console.log(getKeyByValue(contacts, "307-687-2982"));
console.log("");

console.log("Do we know Chris E. Myers' phone number? (yes/no)");
console.log(doesEntryExist(contacts, "Chris E. Myers"));

function getKeyByValue(map, value) {
	for (let [k, v] of map.entries()) {
		if (v === value) {
			return k;
		}
	}
	return null;
}

function doesEntryExist(map, key) {
	return map.has(key);
}
